packages/repo-tools/src/commands/api-reports/api-reports/patchApiReportGeneration.ts

```diff
@@ -0,0 +1,163 @@
+
+            const sourceFile =
+              message.sourceFilePath &&
+              program.getSourceFile(message.sourceFilePath);
+            if (!sourceFile) {
+              throw new Error(
+                `Failed to find source file in program at path "${message.sourceFilePath}"`,
+              );
+            }
+
+            // The local name of the symbol within the file, rather than the exported name
+            let localName = (sourceFile as any).identifiers?.get(symbolName);
+
+            if (!localName) {
+              // Sometimes the symbol name is suffixed with a number to disambiguate,
+              // e.g. "Props_14" instead of "Props" if there are multiple Props interfaces
+              // so we try to strip that suffix and look up the symbol again.
+              const [, trimmedSymbolName] = symbolName.match(/(.*)_\d+/) || [];
+              localName = (sourceFile as any).identifiers?.get(
+                trimmedSymbolName,
+              );
+            }
+
+            if (!localName) {
+              throw new Error(
+                `Unable to find local name of "${symbolName}" in ${sourceFile.fileName}`,
+              );
+            }
+
+            // The local AST node of the export that we're missing
+            const local = (sourceFile as any).locals?.get(localName);
+            if (!local) {
+              return true;
+            }
+
+            // Use the type checker to look up the actual declaration(s) rather than the one in the local file
+            const type = program
+              .getTypeChecker()
+              .getDeclaredTypeOfSymbol(local);
+            if (!type) {
+              throw new Error(
+                `Unable to find type declaration of "${symbolName}" in ${sourceFile.fileName}`,
+              );
+            }
+            const declarations = type.aliasSymbol?.declarations;
+            if (!declarations || declarations.length === 0) {
+              return true;
+            }
+
+            // If any of the TSDoc comments contain a @ignore tag, we ignore this message
+            const isIgnored = declarations.some(declaration => {
+              const tags = [(declaration as any).jsDoc]
+                .flat()
+                .filter(Boolean)
+                .flatMap((tagNode: any) => tagNode.tags);
+
+              return tags.some(tag => tag?.tagName.text === 'ignore');
+            });
+
+            return !isIgnored;
+          });
+        },
+      );
+
+      /**
+       * This monkey patching is for applying prettier to the API reports. This has to be patched into
+       * the middle of the process as API Extractor does a comparison of the contents of the old
+       * and new files during generation. This inserts the formatting just before that comparison.
+       */
+      const content = originalGenerateReviewFileContent.call(
+        this,
+        collector,
+        ...moreArgs,
+      );
+
+      return tryRunPrettier(content);
```

Comment:
@freben applying prettier needs to be sync because of this patching unfortunately, so we may need to stay on the old version or synchronously shell out to the prettier CLI

Reply:
Yeah I know, I've been through poking at this and backed out :) Using the cli - or actually just NOT run prettier at all - were what I thought at the time too.